srtcore/api.cpp

```diff
@@ -649,101 +672,111 @@ int CUDTUnited::newConnection(const SRTSOCKET listen, const sockaddr_any& peer,
-         // the listener set.
-         g->m_listener = ls;
-
-         // Listen on both first connected socket and continued sockets.
-         // This might help with jump-over situations, and in regular continued
-         // sockets the IN event won't be reported anyway.
-         int listener_modes = SRT_EPOLL_ACCEPT | SRT_EPOLL_UPDATE;
-         srt_epoll_add_usock(g->m_RcvEID, ls->m_SocketID, &listener_modes);
-
-         // This listening should be done always when a first connected socket
-         // appears as accepted off the listener. This is for the sake of swait() calls
-         // inside the group receiving and sending functions so that they get
-         // interrupted when a new socket is connected.
-      }
+           // XXX PROLBEM!!! These events are subscribed here so that this is done once, lazily,
+           // but groupwise connections could be accepted from multiple listeners for the same group!
+           // m_listener MUST BE A CONTAINER, NOT POINTER!!!
+           // ALSO: Maybe checking "the same listener" is not necessary as subscruption may be done
+           // multiple times anyway?
+           if (!g->m_listener)
+           {
+               // Newly created group from the listener, which hasn't yet
+               // the listener set.
+               g->m_listener = ls;
+
+               // Listen on both first connected socket and continued sockets.
+               // This might help with jump-over situations, and in regular continued
+               // sockets the IN event won't be reported anyway.
+               int listener_modes = SRT_EPOLL_ACCEPT | SRT_EPOLL_UPDATE;
+               epoll_add_usock_INTERNAL(g->m_RcvEID, ls, &listener_modes);
+
+               // This listening should be done always when a first connected socket
+               // appears as accepted off the listener. This is for the sake of swait() calls
+               // inside the group receiving and sending functions so that they get
+               // interrupted when a new socket is connected.
+           }
 
-      // Add also per-direction subscription for the about-to-be-accepted socket.
-      // Both first accepted socket that makes the group-accept and every next
-      // socket that adds a new link.
-      int read_modes = SRT_EPOLL_IN | SRT_EPOLL_ERR;
-      int write_modes = SRT_EPOLL_OUT | SRT_EPOLL_ERR;
-      srt_epoll_add_usock(g->m_RcvEID, ns->m_SocketID, &read_modes);
-      srt_epoll_add_usock(g->m_SndEID, ns->m_SocketID, &write_modes);
+           // Add also per-direction subscription for the about-to-be-accepted socket.
+           // Both first accepted socket that makes the group-accept and every next
+           // socket that adds a new link.
+           int read_modes = SRT_EPOLL_IN | SRT_EPOLL_ERR;
+           int write_modes = SRT_EPOLL_OUT | SRT_EPOLL_ERR;
+           epoll_add_usock_INTERNAL(g->m_RcvEID, ns, &read_modes);
+           epoll_add_usock_INTERNAL(g->m_SndEID, ns, &write_modes);
```

Comment:
Well, I needed some special marker that should attract attention because of dangerous treats of these functions. Similar thing about `_LOCKED` suffix. This function isn't according to the naming convention either because it looks more like an API function. Probably something like `epollAddSrtSocket` would be better, or even something reaching deeper into the internals, driven out of `CUDT::s_UDTUnited.m_EPoll.add...`

Reply:
There are two naming styles used so far:
- camel case;
- lower case with underscores

I get the point of "eye catching". I am just concerned about adding a third naming style, making the code look like a garland.